src/robusta/core/reporting/blocks.py

```diff
@@ -230,18 +243,51 @@ def __calc_max_width(cls, headers, rendered_rows, table_max_width: int) -> List[
 
         return columns_max_widths
 
+    @classmethod
+    def __trim_rows(cls, contents: str, max_chars: int):
+        # We need to make sure that the total character count doesn't exceed max_chars,
+        # but if we cut off a row in the middle then it messes up the whole table.
+        # So instead remove entire rows at a time
+        if len(contents) <= max_chars:
+            return contents
+
+        truncator = "\n..."
+        max_chars -= len(truncator)
+
+        lines = contents.splitlines()
+        length_so_far = 0
+        lines_to_include = 0
+        for l in lines:
+            new_length = length_so_far + len("\n") + len(l)
+            if new_length > max_chars:
+                break
+            else:
+                length_so_far = new_length
+                lines_to_include += 1
+
+        return "\n".join(lines[:lines_to_include]) + truncator
```

Comment:
We add the `truncator` even in the case we didn't remove any line
Should we?

Reply:
I don't think so. In that case we already left the function in line 252

Reply:
got it